bundles/catalogue/metadata/instance.js

```diff
@@ -0,0 +1,169 @@
+import { showMetadataFlyout } from './view/MetadataFlyout';
+import { MetadataHandler } from './handler/MetadataHandler';
+
+Oskari.clazz.define('Oskari.catalogue.bundle.metadata.MetadataBundleInstance',
+    function () {
+        this.sandbox = null;
+        this.started = false;
+        this.requestHandler = null;
+        this.flyoutControls = null;
+        this.layerService = null;
+        this.loc = Oskari.getMsg.bind(null, this.getName());
+        this.handler = new MetadataHandler(this);
+        this.handler.addStateListener(state => this.onStateUpdate(state));
+    }, {
+        __name: 'catalogue.metadata',
+
+        getName: function () {
+            return this.__name;
+        },
+
+        getSandbox: function () {
+            return this.sandbox;
+        },
+
+        start: function (sandbox) {
+            if (this.started) {
+                return;
+            }
+            this.started = true;
+
+            this.sandbox = sandbox || Oskari.getSandbox(this.conf.sandbox || 'sandbox');
+            this.mapmodule = this.sandbox.findRegisteredModuleInstance('MainMapModule');
+
+            this.sandbox.register(this);
+
+            this.requestHandler = Oskari.clazz.create('Oskari.catalogue.bundle.metadata.request.ShowMetadataRequestHandler', this);
+            this.sandbox.requestHandler('catalogue.ShowMetadataRequest', this.requestHandler);
+
+            Object.getOwnPropertyNames(this.eventHandlers).forEach(p => this.sandbox.registerForEventByName(this, p));
+
```

Comment:
I think removing the stateful from this might be a good idea, but we need to handle state anyway on startup to make sure we can show the requested metadata on startup. There's an URL-param that can be passed to this as state and it's expected to be shown on startup if it receives the uuid on state.